pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -83,21 +83,22 @@ def _check_consider_iterating_dictionary(self, node: nodes.Call) -> None:
             return
         if node.func.attrname != "keys":
             return
+        comp_ancestor = utils.get_node_first_ancestor_of_type(node, nodes.Compare)
         if (
             isinstance(node.parent, (nodes.For, nodes.Comprehension))
-            or isinstance(node.parent, nodes.Compare)
+            or isinstance(comp_ancestor, nodes.Compare)
             and any(
                 op
-                for op, comparator in node.parent.ops
-                if op == "in" and comparator is node
+                for op, comparator in comp_ancestor.ops
+                if op in ["in", "not in"]
+                and (comparator in node.node_ancestors() or comparator is node)
```

Comment:
Just to check: is this ever untrue? I'm not sure how the `comparator` of `comp_ancestor.ops` works, but I can imagine this is always true?

Reply:
it could be a comparison like 

`if a == {}.keys() in {}`

one of the tests fail if this check is omitted, see L68 of test file 

`VAR = [1, 2] == {}.keys() in {False}`